### PR TITLE
Updates packages to satisfy vulnerability scanner

### PR DESCRIPTION
**Pull request by thomasbrockmeier-ams**
Vulnerability scanner reports for [postprocessing](https://portal.azure.com/#view/Microsoft_Azure_Security_CloudNativeCompute/ImageScanReportBlade/digest/sha256%3A6ec368c69ed4b50f9a5ebcc601594c64301ea1588a401d91a1062649dccbded7/registryHost/cvtweuacrogidgmnhwma3zq.azurecr.io/repository/postprocessing/registry/cvtweuacrogidgmnhwma3zq/subscriptionId/5951db82-f968-41fb-b67c-6a78f1be5270/resourceGroup/cvt-weu-o-rg/securityStateSeverity~/4/openedFromExternalLink~/false/os/Linux) and [retrieve](https://portal.azure.com/#view/Microsoft_Azure_Security_CloudNativeCompute/ImageScanReportBlade/digest/sha256%3A1bf9f9ff6e8928398fd4506a38fce79385d2e8248a957120b4b18f618a259b6f/registryHost/cvtweuacrogidgmnhwma3zq.azurecr.io/repository/retrieve/registry/cvtweuacrogidgmnhwma3zq/subscriptionId/5951db82-f968-41fb-b67c-6a78f1be5270/resourceGroup/cvt-weu-o-rg/securityStateSeverity~/4/openedFromExternalLink~/false/os/Linux) indicated unsafe dependencies.

Bumped azure-cli, numpy, and Geolocalization_of_Street_Objects (which contains the Pillow dependency). Geolocalization_of_Street_Objects was fixed in https://github.com/Computer-Vision-Team-Amsterdam/Geolocalization_of_Street_Objects/pull/5, but dependency reference was not yet updated :)

### CODE REVIEW

File: postprocessing.Dockerfile [hunk 1]
```diff
@@ -22,20 +22,20 @@ RUN pip install --no-cache \
...
         shapely==1.8.2 \
         tqdm==4.64.0 \
-        azure-cli==2.39.0 \
+        azure-cli==2.42.0 \
         azure-identity==1.10.0 \
         azure-keyvault-secrets==4.5.1 \
         azure-storage-blob==12.13.1 \
-        git+https://git@github.com/Computer-Vision-Team-Amsterdam/Geolocalization_of_Street_Objects.git@faba8f6f4a94e545135dd24aea398defe2c69f97 \
-        git+https://git@github.com/Computer-Vision-Team-Amsterdam/panorama.git@v0.2.3
+        git+https://git@github.com/Computer-Vision-Team-Amsterdam/Geolocalization_of_Street_Objects.git@17d2945ed46d40bd770ac43e0cf9f5e74f36e9ef \
```
Comment (epureanudiana) [inline postprocessing.Dockerfile:37]: should we also use release tags for the geolocalization package?

File: retrieve_images.Dockerfile [hunk 2]
```diff
@@ -8,7 +8,7 @@ ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 RUN pip install \
     requests==2.26.0 \
-    azure-cli==2.39.0 \
+    azure-cli==2.42.0 \
```
Comment (epureanudiana) [inline retrieve_images.Dockerfile:11]: I think azure-cli is not even needed in the Dockerfile

Comment (thomasbrockmeier-ams) [reply retrieve_images.Dockerfile:11]: There is an import `from azure_storage_utils import StorageAzureClient` in postprocessing.py that has a further dependency on `azure` packages:

```
from azure.core.exceptions import ResourceNotFoundError
from azure.identity import ManagedIdentityCredential
from azure.keyvault.secrets import SecretClient
from azure.storage.blob import BlobServiceClient
```

These should come from the `azure-cli` package and its add-ons, right?

File: postprocessing.Dockerfile (hunk 1 above)
Comment (thomasbrockmeier-ams) [reply postprocessing.Dockerfile:37]: Good idea, that would make this a bit more readable. I'll tag it right now

**Pull request merged**